source/NVDAObjects/UIA/__init__.py

```diff
@@ -1016,7 +1016,7 @@ def kwargsFromSuper(cls,kwargs,relation=None):
 				log.debugWarning("getFocusedElement failed", exc_info=True)
 				return False
 			# Ignore this object if it is non native.
-			if not UIAHandler.handler.isNativeUIAElement(UIAElement):
+			if not UIAHandler.handler.isNativeUIAElement(UIAElement) and ignoreNonNativeElementsWithFocus:

```

Comment:
Please check ignoreNonnativeElementWithFocus before calling isNativeUIAElement, as the isNativeUIAElement call is more costly.